### DISCUSSION TITLE
Can a message be BDS44 and BDS45 at the same time?

### DISCUSSION THREAD

**Discussion by ElSabio97** [junzis/pyModeS#118] [Q&A]
Hi @junzis !!

It has been a long time hehe. I am giving a friend a helping hand trying to plot a map of temperatures.

We are using the BDS44 and BDS45 capabilities you have coded for pyModeS. We first download a chunk of data using `traffic` and then filter out the BDS44/45 messages. Like this:

```
import pyModeS as pms
from traffic.data import opensky, eurofirs
import pandas as pd

dia = pd.to_datetime("2021-09-20 12:00") #Fecha y hora de comienzo del vuelo.
delta = 5 #Duración en horas (solo valen números enteros).

frontera = eurofirs['LSAS']

vuelo = opensky.history( #Saca datos ADSB.
    (dia).strftime('%Y-%m-%d %H:%M:%S')[0:19],
    (dia + pd.Timedelta(minutes=delta)).strftime('%Y-%m-%d %H:%M:%S')[0:19],
    bounds=frontera
)

df = opensky.extended(
    (dia).strftime('%Y-%m-%d %H:%M:%S')[0:19],
    (dia + pd.Timedelta(minutes=delta)).strftime('%Y-%m-%d %H:%M:%S')[0:19],
    bounds=frontera
)

df.data['BDS'] = df.data['rawmsg'].apply(lambda x: pms.bds.infer(x, True))

df.data = df.data[(df.data.BDS.str.contains('BDS44')) | (df.data.BDS.str.contains('BDS45'))]

df.data['BDS44'] = df.data['rawmsg'].apply(lambda x: pms.bds.bds44.is44(x))
df.data['BDS45'] = df.data['rawmsg'].apply(lambda x: pms.bds.bds45.is45(x))
```

The thing here is that we have found a message that is infered as both BDS44 and BDS45. That is cool, but when we pass `is44` and `is45`, both come out as true.

If you do not want to run that code, this is the message: `a000133800000030ac0000e35b83`

Have a nice day!

**Comment by junzis** [comment]
Hi @ElSabio97, unfortunately, in this case, there is no way to tell whether this is BDS 44 or 45. 

Nevertheless, BDS 45 messages are much rarer than BDS 44 messages.

**Reply by ElSabio97** [reply]
How would you try to discard this case (s)? 

Should I just not consider them?

**Reply by junzis** [reply]
Yes, that's what I would do in most cases.
